x-pack/plugins/session_view/public/components/ProcessTree/index.tsx

```diff
@@ -0,0 +1,184 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useRef, useEffect, useLayoutEffect, useCallback } from 'react';
+import { EuiButton } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { ProcessTreeNode } from '../ProcessTreeNode';
+import { useProcessTree } from './hooks';
+import { Process, ProcessEventsPage, ProcessEvent } from '../../../common/types/process_tree';
+import { useScroll } from '../../hooks/use_scroll';
+import { useStyles } from './styles';
+
+type FetchFunction = () => void;
+
+interface ProcessTreeDeps {
+  // process.entity_id to act as root node (typically a session (or entry session) leader).
+  sessionEntityId: string;
+
+  data: ProcessEventsPage[];
+
+  jumpToEvent?: ProcessEvent;
+  isFetching: boolean;
+  hasNextPage: boolean | undefined;
+  hasPreviousPage: boolean | undefined;
+  fetchNextPage: FetchFunction;
+  fetchPreviousPage: FetchFunction;
+
+  // plain text search query (only searches "process.working_directory process.args.join(' ')"
+  searchQuery?: string;
+
+  // currently selected process
+  selectedProcess?: Process | null;
+  onProcessSelected: (process: Process) => void;
+  setSearchResults?: (results: Process[]) => void;
+}
+
+export const ProcessTree = ({
+  sessionEntityId,
+  data,
+  jumpToEvent,
+  isFetching,
+  hasNextPage,
+  hasPreviousPage,
+  fetchNextPage,
+  fetchPreviousPage,
+  searchQuery,
+  selectedProcess,
+  onProcessSelected,
+  setSearchResults,
+}: ProcessTreeDeps) => {
+  const styles = useStyles();
+
+  const { sessionLeader, processMap, searchResults } = useProcessTree({
+    sessionEntityId,
+    data,
+    searchQuery,
+  });
+
+  const scrollerRef = useRef<HTMLDivElement>(null);
+  const selectionAreaRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (setSearchResults) {
+      setSearchResults(searchResults);
+    }
+  }, [searchResults, setSearchResults]);
+
+  useScroll({
+    div: scrollerRef.current,
+    handler: (pos: number, endReached: boolean) => {
+      if (!isFetching && endReached) {
+        fetchNextPage();
+      }
+    },
+  });
+
+  /**
+   * highlights a process in the tree
+   * we do it this way to avoid state changes on potentially thousands of <Process> components
+   */
+  const selectProcess = useCallback((process: Process) => {
+    if (!selectionAreaRef || !scrollerRef) {
+      return;
+    }
+
+    if (!selectionAreaRef.current || !scrollerRef.current) {
+      return;
+    }
+
+    const selectionAreaEl = selectionAreaRef.current;
+    selectionAreaEl.style.display = 'block';
+
+    // TODO: concept of alert level unknown wrt to elastic security
+    const alertLevel = process.getMaxAlertLevel();
+
+    if (alertLevel && alertLevel >= 0) {
+      selectionAreaEl.style.backgroundColor =
+        alertLevel > 0 ? 'rgba(229, 115, 115, 0.24)' : '#F2C94C4A';
```

Comment:
Are there EUI constants for these colors?

Reply:
good catch, will check with Design team